### PR TITLE
Replace bulk archive script with management command

### PR DESCRIPTION
**Pull request by willbarton**
This change replaces the bulk archive script with a management command that can take multiple paths and can archive based on year, month, and day arguments based on either first or last published date.

By default this command will set pages to archived if their `first_published_at` date is 2 years or more from when the script is run.

See #6041 for more context on the bulk archive script.

To archive content under the Blog, Newsroom, and Research & Reports filterable lists, one would run the following command:

```
./cfgov/manage.py archive_pages about-us/blog about-us/newsroom data-research/research-reports
```

And see the following output:

```
Found 992 pages within Blog older than 2018-11-04 16:09:06.227133+00:00
Archived 992 pages under Blog
Found 1070 pages within Newsroom older than 2018-11-04 16:09:06.227133+00:00
Archived 1070 pages under Newsroom
Found 288 pages within Research & Reports older than 2018-11-04 16:09:06.227133+00:00
Archived 288 pages under Research & Reports
```

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)

### CODE REVIEW

File: cfgov/v1/management/commands/archive_pages.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        # Get the current date/time and then get our cutoff date for archiving
+        # based on it.
+        archived_at = timezone.now()
+        cutoff_date = archived_at - relativedelta(
+            years=options["years"],
+            months=options["months"],
+            days=options["days"],
+        )
+
+        # Construct a Q object to filter on based on this command-line argument
```
Comment (Scotchester) [inline cfgov/v1/management/commands/archive_pages.py:80]: Trivial: consistency of ending complete sentence comments with a period.

```suggestion
        # Construct a Q object to filter on based on this command-line argument.
```

File: cfgov/v1/management/commands/archive_pages.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        if options["by_published_date"] == "last":
+            published_date_filter = Q(last_published_at__lt=cutoff_date)
+        else:
+            published_date_filter = Q(first_published_at__lt=cutoff_date)
+
+        # We"ll use Wagtail"s page routing to resolve the page at the given the
+        # URL paths
```
Comment (Scotchester) [inline cfgov/v1/management/commands/archive_pages.py:87]:
```suggestion
        # We'll use Wagtail's page routing to resolve the page at the given the
        # URL paths.
```

File: cfgov/v1/management/commands/archive_pages.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        default_site = Site.objects.get(is_default_site=True)
+        root_page = default_site.root_page
+
+        for path in url_paths:
+            path_components = path.split("/")
+
+            # Get the filterable list page we"re interested in
```
Comment (Scotchester) [inline cfgov/v1/management/commands/archive_pages.py:94]:
```suggestion
            # Get the filterable list page we're interested in.
```

File: cfgov/v1/management/commands/archive_pages.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+            try:
+                filterable_page = root_page.route(None, path_components).page
+            except Http404:
+                raise CommandError(
+                    f"Unable to find a filterable list page for {path}. "
+                    "Ensure that the path is correct, and leave off any "
+                    "leading or trailing / characters."
+                )
+
+            # Get the filterable list queryset and filter it
```
Comment (Scotchester) [inline cfgov/v1/management/commands/archive_pages.py:104]:
```suggestion
            # Get the filterable list QuerySet and filter it.
```

File: cfgov/v1/management/commands/archive_pages.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+        for path in url_paths:
+            path_components = path.split("/")
+
+            # Get the filterable list page we"re interested in
+            try:
+                filterable_page = root_page.route(None, path_components).page
+            except Http404:
+                raise CommandError(
+                    f"Unable to find a filterable list page for {path}. "
```
Comment (Scotchester) [inline cfgov/v1/management/commands/archive_pages.py:99]: This suggestion makes the error message more accurate, but the better fix is probably to check the page type if it doesn't 404. If the page is found but is not a filterable page, it throws "AttributeError: 'LandingPage' object has no attribute 'get_filterable_queryset'".

```suggestion
                    f"Unable to find a page at {path}. "
```

File: cfgov/v1/management/commands/archive_pages.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+            filtered_pages = filterable_page.get_filterable_queryset().filter(
+                published_date_filter,
+                is_archived="no",
+            )
+
+            # Archive the content, letting the user know the title of the
+            # filterable list page, the cuttoff date and how many pages will be
+            # archived.
+            self.stdout.write(
+                f"Found {filtered_pages.count()} pages within "
```
Comment (Scotchester) [inline cfgov/v1/management/commands/archive_pages.py:114]:
```suggestion
                f"Found {filtered_pages.count()} unarchived pages within "
```

File: cfgov/v1/management/commands/archive_pages.py [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+                f"{filterable_page.title} older than {cutoff_date}"
+            )
+            with transaction.atomic():
+                update_count = filtered_pages.select_for_update().update(
+                    is_archived="yes",
+                    archived_at=archived_at
+                )
+            self.stdout.write(
+                f"Archived {update_count} pages under {filterable_page.title}"
+            )
```
Comment (Scotchester) [inline cfgov/v1/management/commands/archive_pages.py:124]: How would you feel about combining the output into a single "Found and archived _n_ unarchived pages within <parent> older than <cutoff_date>" message?

File: cfgov/v1/management/commands/archive_pages.py [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+                published_date_filter,
+                is_archived="no",
+            )
+
+            # Archive the content, letting the user know the title of the
+            # filterable list page, the cuttoff date and how many pages will be
+            # archived.
+            self.stdout.write(
+                f"Found {filtered_pages.count()} pages within "
+                f"{filterable_page.title} older than {cutoff_date}"
```
Comment (Scotchester) [inline cfgov/v1/management/commands/archive_pages.py:115]: Nice to have: Format the date in ET (or server local time) and without the partial seconds.

**Review by Scotchester — CHANGES_REQUESTED** [review]
This is looking great! A few minor comments and suggestions.

One more: I wonder if it's worth adding a test for the 2nd or more run that would include previously-archived pages. No concerns about that when testing locally, just to make sure that no side effects of multiple runs crop up in the future.

File: cfgov/v1/management/commands/archive_pages.py (hunk 1 above)
Comment (willbarton) [reply cfgov/v1/management/commands/archive_pages.py]: Done in 1e70b3164.

File: cfgov/v1/management/commands/archive_pages.py (hunk 2 above)
Comment (willbarton) [reply cfgov/v1/management/commands/archive_pages.py]: Done in 1e70b31.

File: cfgov/v1/management/commands/archive_pages.py (hunk 3 above)
Comment (willbarton) [reply cfgov/v1/management/commands/archive_pages.py]: Done in 1e70b31.

File: cfgov/v1/management/commands/archive_pages.py (hunk 5 above)
Comment (willbarton) [reply cfgov/v1/management/commands/archive_pages.py]: Done in 1e70b31.

File: cfgov/v1/management/commands/archive_pages.py (hunk 6 above)
Comment (willbarton) [reply cfgov/v1/management/commands/archive_pages.py]: Done in 1e70b31.

File: cfgov/v1/management/commands/archive_pages.py (hunk 7 above)
Comment (willbarton) [reply cfgov/v1/management/commands/archive_pages.py]: Done in 1e70b31.
It now appears like:

```
Found and archived 994 pages within Blog older than 2018-11-09 14:26 UTC.
```

File: cfgov/v1/management/commands/archive_pages.py (hunk 8 above)
Comment (willbarton) [reply cfgov/v1/management/commands/archive_pages.py]: Done in 1e70b31.
It now appears like:

```
Found and archived 994 pages within Blog older than 2018-11-09 14:26 UTC.
```

**Review by Scotchester — APPROVED** [review]
Changes all look good to me!

**Pull request merged**